lib/src/main/java/org/exoplatform/social/addons/storage/entity/ActivityEntity.java

```diff
@@ -57,41 +57,33 @@
                 query = "SELECT a FROM SocActivity a WHERE a.hidden = false  AND a.updatedDate > :sinceTime AND exists (SELECT item FROM a.streamItems item WHERE item.ownerId in (:owner)) ORDER BY a.updatedDate ASC"),
         @NamedQuery(name = "SocActivity.getActivityByOwnerAndProviderId",
                 query = "SELECT a FROM SocActivity a WHERE a.hidden = false AND a.providerId = :providerId AND exists (SELECT item FROM a.streamItems item WHERE item.ownerId in (:owner)) ORDER BY a.updatedDate DESC"),
+        @NamedQuery(name = "SocActivity.getConnections",
+                query = "SELECT conn1.receiver.id from SocConnection conn1 where conn1.status = :connStatus  and conn1.receiver.id = :identityId" +
+                        " union " +
+                        "SELECT conn2.sender.id from SocConnection conn2 where conn2.status = :connStatus and conn2.sender.id = :identityId "),
```

Comment:
We should put this query on ConnectionEntity class, I think. It's easier to maintain and follow. We should rename it to "SocConnection.getConnections" also.

If I remember well, JPA 2.0 (the version we are using) does not support union in query. We could use this kind of query with Hibernate implementation?